pkg/dependencies/dependents.go

```diff
@@ -0,0 +1,194 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencies
+
+import (
+	"context"
+	"fmt"
+	"github.com/Khan/genqlient/graphql"
+	"github.com/guacsec/guac/cmd/guacone/cmd"
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"github.com/guacsec/guac/pkg/handler/collector/deps_dev"
+	"sort"
+)
+
+type dependencyNode struct {
+	dependents map[string]bool // map of the Name of the dependent to whether it is a dependent of the current node
+}
+
+type PackageName struct {
+	Name           string
+	DependentCount int
+}
+
+type Dependencies interface {
+	GetSortedDependents() ([]PackageName, error)
+}
+
+type dependencies struct {
+	ctx       context.Context
+	gqlClient graphql.Client
+}
+
+func New(ctx context.Context, gqlClient graphql.Client) Dependencies {
+	return &dependencies{
+		gqlClient: gqlClient,
+		ctx:       ctx,
+	}
+}
+
+func (deps *dependencies) GetSortedDependents() ([]PackageName, error) {
+	packages, err := findAllDependents(deps.gqlClient)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get dependents: %v", err)
+	}
+
+	var packagesArr []PackageName
+
+	for n, d := range packages {
+		packagesArr = append(packagesArr, PackageName{Name: n, DependentCount: len(d.dependents)})
+	}
+
+	sort.Slice(packagesArr, func(i, j int) bool {
+		return packagesArr[i].DependentCount > packagesArr[j].DependentCount
+	})
+	return packagesArr, nil
+}
+
+// findAllDependents queries a GraphQL endpoint to find all dependencies and constructs a map of dependencyNode.
+func findAllDependents(gqlClient graphql.Client) (map[string]dependencyNode, error) {
+	ctx := context.Background()
```

Comment:
ctx should be passed in, not background.

Reply:
Taking `ctx` as a parameter.